packages/dds/tree/src/test/cursorTestSuite.ts

```diff
@@ -117,18 +118,29 @@ export const cursorTestCases: [string, JsonableTree][] = [
  * @param factory - Creates the cursor to be tested with or without provided data.
  * @param dataFromCursor - Gets a JsonableTree from the provided cursor.
  */
-export function testJsonableTreeCursor(
+export function testJsonableTreeCursor<T>(
     cursorName: string,
-    factory: (data: JsonableTree) => ITreeCursor,
-    dataFromCursor: (cursor: ITreeCursor) => JsonableTree,
+    cursorFactory: (data: T) => ITreeCursor,
+    dataFromCursor: (cursor: ITreeCursor) => T,
     testRooted = true,
 ): void {
+    function dataFromJsonableTree(data: JsonableTree): T {
+        // Use text cursor to provide input data
+        return dataFromCursor(singleTextCursor(data));
+    }
+    function factory(data: JsonableTree): ITreeCursor {
+        return cursorFactory(dataFromJsonableTree(data));
+    }
```

Comment:
Nit: I personally like white space. Feel free to ignore this comment though.

```suggestion
    }
    
    function factory(data: JsonableTree): ITreeCursor {
        return cursorFactory(dataFromJsonableTree(data));
    }
    
```

Reply:
Your suggestion has training white space on the blank line.
```suggestion
    }

    function factory(data: JsonableTree): ITreeCursor {
        return cursorFactory(dataFromJsonableTree(data));
    }
```

Reply:
Oh whoops, nice catch.